app/multichain-accounts/discovery.ts

```diff
@@ -2,8 +2,18 @@ import { Bip44Account } from '@metamask/account-api';
 import { EntropySourceId, KeyringAccount } from '@metamask/keyring-api';
 import { MultichainAccountWallet } from '@metamask/multichain-account-service';
 import Engine from '../core/Engine';
+import { trace, TraceOperation, TraceName } from '../util/trace';
 
-export async function discoverAndCreateAccounts(
+/**
+ * Discover and create accounts.
+ *
+ * This function is wrapped by {@link discoverAndCreateAccounts} to add tracing
+ * and should not be called directly.
+ *
+ * @param entropySource - Entropy source ID to use for account discovery
+ * @returns The number of discovered and created accounts.
+ */
+async function _discoverAndCreateAccounts(
```

Comment:
```suggestion
async function #discoverAndCreateAccounts(
```

Should it be marked as private, somehow?

Reply:
I think it's not possible to use `#` outside a class body. I made it module private by not exporting it.